robot/resources/lib/python_keywords/http_gate.py

```diff
@@ -286,10 +286,10 @@ def get_object_and_verify_hashes(
         nodes=nodes,
     )
     # for some reason we can face with case when nodes_list is empty due to object resides in all nodes
-    if nodes_list is None:
-        random_node = random.choice(nodes)
-    else:
+    if nodes_list:
         random_node = random.choice(nodes_list)
+    else:
```

Comment:
This may break the initial intention of this method.
If someone expecting that this method MUST choose nodes without object and we silently ignoring that error may become a hidden issue.